std/experimental/color/package.d

```diff
@@ -0,0 +1,500 @@
+    static assert(colorFromString("0xFF8000") == RGB8(0xFF, 0x80, 0x00));
+
+    // 4/8 digita (/w alpha)
+    static assert(colorFromString!RGBA8("#8C41") == RGBA8(0xCC, 0x44, 0x11, 0x88));
+    static assert(colorFromString!RGBA8("#80CC4401") == RGBA8(0xCC, 0x44, 0x01, 0x80));
+
+    // named colors (case-insensitive)
+    static assert(colorFromString("red") == RGB8(0xFF, 0x0, 0x0));
+    static assert(colorFromString("WHITE") == RGB8(0xFF, 0xFF, 0xFF));
+    static assert(colorFromString("LightGoldenrodYellow") == RGB8(250,250,210));
+}
+
+
+package:
+
+import std.traits : isInstanceOf, TemplateOf;
+import std.typetuple : TypeTuple;
+
+
+RGBA8 colorFromStringImpl(const(char)[] str) pure @safe
+{
+    static const(char)[] getHex(const(char)[] hex) pure nothrow @nogc @safe
+    {
+        if (hex.length > 0 && (hex[0] == '#' || hex[0] == '$'))
+            hex = hex[1..$];
+        else if (hex.length > 1 && (hex[0] == '0' && hex[1] == 'x'))
+            hex = hex[2..$];
+        foreach (i; 0 .. hex.length)
+        {
+            if (!(hex[i] >= '0' && hex[i] <= '9' || hex[i] >= 'a' && hex[i] <= 'f' || hex[i] >= 'A' && hex[i] <= 'F'))
+                return null;
+        }
+        return hex;
+    }
+
+    const(char)[] hex = getHex(str);
+    if (hex)
+    {
+        static ubyte val(char c) pure nothrow @nogc @safe
+        {
+            if (c >= '0' && c <= '9')
+                return cast(ubyte)(c - '0');
+            else if (c >= 'a' && c <= 'f')
+                return cast(ubyte)(c - 'a' + 10);
+            else
+                return cast(ubyte)(c - 'A' + 10);
+        }
+
+        if (hex.length == 3)
+        {
+            ubyte r = val(hex[0]);
+            ubyte g = val(hex[1]);
+            ubyte b = val(hex[2]);
+            return RGBA8(cast(ubyte)(r | (r << 4)), cast(ubyte)(g | (g << 4)), cast(ubyte)(b | (b << 4)), 0);
+        }
+        if (hex.length == 4)
+        {
+            ubyte a = val(hex[0]);
+            ubyte r = val(hex[1]);
+            ubyte g = val(hex[2]);
+            ubyte b = val(hex[3]);
+            return RGBA8(cast(ubyte)(r | (r << 4)), cast(ubyte)(g | (g << 4)), cast(ubyte)(b | (b << 4)), cast(ubyte)(a | (a << 4)));
+        }
+        if (hex.length == 6)
+        {
+            ubyte r = cast(ubyte)(val(hex[0]) << 4) | val(hex[1]);
+            ubyte g = cast(ubyte)(val(hex[2]) << 4) | val(hex[3]);
+            ubyte b = cast(ubyte)(val(hex[4]) << 4) | val(hex[5]);
+            return RGBA8(r, g, b, 0);
+        }
+        if (hex.length == 8)
+        {
+            ubyte a = cast(ubyte)(val(hex[0]) << 4) | val(hex[1]);
+            ubyte r = cast(ubyte)(val(hex[2]) << 4) | val(hex[3]);
+            ubyte g = cast(ubyte)(val(hex[4]) << 4) | val(hex[5]);
+            ubyte b = cast(ubyte)(val(hex[6]) << 4) | val(hex[7]);
+            return RGBA8(r, g, b, a);
+        }
+
+        throw new Exception("Hex string has invalid length");
```

Comment:
You could throw a `std.conv.ConvException` here. Also, it seems that you only throw two times in the whole package. How about chaging the signature to 
```D
Color colorFromString(Color = RGB8)(const(char)[] str, Color default = Color(0,0,0,0))
pure @safe nothrow
```

On conversion failure the result would be set to the default value. Users would be able to throw themselves when it's really a problem. Another options would be

```D
Color colorFromString(Color = RGB8)(const(char)[] str, out bool failed) 
pure @safe nothrow
```

?